### ISSUE TITLE
Upgrade script: check it has been modified correctly as part of release build

### ISSUE THREAD

**Issue by Tom-Willemsen** [ISISComputingGroup/IBEX#4708]
The [upgrade script update](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Config-Upgrader#creating-a-production-upgrade-script) is prone to mistakes when we are building a new release.

As a developer, I would like the release build to "fail-fast" if the upgrade script is not correct. The jenkins build should be able to detect that it is on a branch called `Release_x.x.x` and verify that the config update script agrees with this version number. This test should be run *before* the main epics build, so that a mistake can be spotted and corrected quickly.

This should be possible to run as a simple python script, which can then be called from `EPICS/Jenkinsfile_release`

Alternatively, if we can automatically update the upgrade script correctly as part of the release build this would be even better (but probably harder).

so basic idea:

- a Jenkins release build will be run which used the [ `Jenkinsfile_release` ](https://github.com/ISISComputingGroup/EPICS/blob/master/Jenkinsfile_release) configuration file
- This job will be running checked out on a  `Release_x.x.x`  branch, this can be accesed via `${env.BRANCH_NAME}` in the file
- If we are doing a build for e.g. Release_12.0.1 there should be a line ("12.0.1", None) as the last entry in the file  https://github.com/ISISComputingGroup/EPICS-upgrade/blob/fdb07a56313123146010148d54038ef6dfb651e0/upgrade.py
- the purpose of the ticket is to add a check to Jenkinsfile_release that confirms upgrade.py is correct
- it should be possible to create a python script e.g. check_upgrade.py in the same area as upgrade.py that loads in the UPGRADE_STEPS and confirms the entries. As `__main__`  is protected you can probably directly import `upgrade.py` and then check the last entry in the UPGRADE_STEPS list is the expected value
-  So to summarise: Jenkinsfile_release will call check_upgrade.py passing it the expected version, this script will then import UPGRADE_STEPS and check versions agree. If it doesn't, build will be failed.

**Comment by FreddieAkeroyd** [comment]
@danielmaclaren @esmith1729 have added extra notes to description

**Comment by danielmaclaren** [comment]
Links to pull requests: https://github.com/ISISComputingGroup/EPICS-upgrade/pull/74
